docs/verifiable-document/document-template.md

```diff
@@ -4,8 +4,285 @@ title: Deploying Document Renderer
+import { object } from "@storybook/addon-knobs";
+import { CustomTemplate } from "./customTemplate";
+import { customTemplateCertificate } from "../sample";
+
+<Meta title="MDX|COC Template" component={CustomTemplate} />
+
+# CustomTemplate component
+
+<Description of={CustomTemplate} />
+
+# Props
+
+<Props of={CustomTemplate} />
+
+# Usage
+
+<Preview>
+  <Story name="basic sample">
+    <CustomTemplate document={object("document", customTemplateCertificate)} />
+  </Story>
+</Preview>
+```
+
+### Update sample document data and type
+
+To update the raw document data and the corresponding data type, you will need to update the data definition file in `src/templates/coc/samples.ts`:
+
+```js
+import { Document } from "@govtechsg/decentralized-renderer-react-components";
+
+export interface CustomTemplateCertificate extends Document {
+  name: string;
+  recipient: {
+    name: string
+  };
+}
+
+export const customTemplateCertificate: CustomTemplateCertificate = {
+  name: "OpenAttestation Tutorial Certificate of Completion",
+  recipient: {
+    name: "John Doe"
+  },
+  $template: {
+    name: "custom",
+    type: "EMBEDDED_RENDERER",
+    url: "http://localhost:3000"
+  }
+};
+```
+
+### Registering the COC template
+
+`src/templates/index.tsx` is a directory of all the templates available in this renderer.
+
+To register a new template, simply add it as a key to the `registry` constant. Take note that the key will be the `$template.name` value defined in the document data.
+
+Replace `src/templates/index.tsx` with the following code to add the new `COC` template:
+
+```js
+import { TemplateRegistry } from "@govtechsg/decentralized-renderer-react-components";
+import { templates as defaultTemplate } from "./customTemplate";
+import { templates as cocTemplate } from "./coc";
+
+export const registry: TemplateRegistry<any> = {
+  default: defaultTemplate,
+  COC: cocTemplate
+};
+```
+
+### Multiple views for a Template
+
+An OA document may have multiple views, each of them rendered in a separate tabs. The views are defined in the `templates` array in `index.tsx` of the template folder. Each of these views can have its own id, label and template code.
+
+For our Certificate of Completion, we will only use a single view. So we will remove any additional views and rename the first view's label as "Certificate".
+
+Replace the code in `src/templates/coc/index.tsx` with the following:
+
+```js
+import { CustomTemplate } from "./customTemplate";
+
+export const templates = [
+  {
+    id: "certificate",
+    label: "Certificate",
+    template: CustomTemplate
+  }
+];
+```
+
+### Developing the COC Template View
+
+Finally, once all the components have been wired up, we may proceed to style our Certificate of Completion.
+
+To change how the data is being renderered, we simply create a React component that takes in the raw document in the `document` props and render the corresponding HTML code.
```

Comment:
I think "raw document" might be confusing as there are 2 properties, one called document (which contain initial data of the document before wrapping up) and raw document which is the wrapped document

Reply:
This comment is for any reference using raw document data

Reply:
ok no more raw documents